src/ucp/core/ucp_worker.c

```diff
@@ -637,21 +663,20 @@ ucp_worker_iface_error_handler(void *arg, uct_ep_h uct_ep, ucs_status_t status)
         goto out;
     }
 
-    /* TODO: need to optimize uct_ep -> ucp_ep lookup */
-    ucs_list_for_each(ep_ext, &worker->all_eps, ep_list) {
-        ucp_ep = ucp_ep_from_ext_gen(ep_ext);
-        lane = ucp_ep_lookup_lane(ucp_ep, uct_ep);
-        if (lane != UCP_NULL_LANE) {
-            ret_status = ucp_worker_iface_handle_uct_ep_failure(ucp_ep, lane,
-                                                                uct_ep, status);
-            goto out;
-        }
+    ucp_ep = ucp_worker_find_lane(&worker->all_eps, uct_ep, &lane);
+    if (ucp_ep == NULL) {
+        ucp_ep = ucp_worker_find_lane(&worker->internal_eps, uct_ep, &lane);
     }
 
-    ucs_error("worker %p: uct_ep %p isn't associated with any ucp endpoint and "
-              "was not scheduled to be discarded",
-              worker, uct_ep);
-    ret_status = UCS_ERR_NO_ELEM;
+    if (ucp_ep != NULL) {
+        ret_status = ucp_worker_iface_handle_uct_ep_failure(ucp_ep, lane,
+                                                            uct_ep, status);
+    } else {
+        ucs_error("worker %p: uct_ep %p isn't associated with any ucp endpoint"
+                  "and was not scheduled to be discarded",
```

Comment:
please add space between the parts of this trace

Reply:
done